### PR TITLE
Feat memory

### LINKED ISSUE #22: expose logs around the service memory footprint
## Description

The aim of this ticket is to setup tooling which allows for realtime check around the amount of memory the nest service is taking up.

### PR DESCRIPTION
**Pull request by Zolo-Ryan**
## Description
This pr is to monitor the memory usage of our service.

### Issue: #22

### CODE REVIEW

**Comment by Zolo-Ryan** [comment]
Please review only: src/memory-monitor/memory-monitor.service.ts
since this branch is created from another work in progress branch and other changed files in this pull request will be reviewed in other pr.

File: src/app.service.ts [hunk 1]
```diff
@@ -72,7 +73,7 @@ export class AppService {
     };
     const response = await axios.request(reqOptions);
```
Comment (BackendBananeWala) [inline src/app.service.ts:74]: Can we build a wrapper around axios calls if are hitting the same host?

File: src/app.service.ts [hunk 2]
```diff
@@ -141,19 +142,21 @@ export class AppService {
...
+    // const user = await this.prismaService.user.findUnique({
+    //   where: {
+    //     username: username,
+    //     email: email,
+    //   },
+    // });
+    const user = null;
     if (user) {
       throw new ConflictException({
         error: 'Duplicate entry',
```
Comment (BackendBananeWala) [inline src/app.service.ts:154]: Can we use Non Spaced error codes? Like DUPLICATE_ERROR. So we wont have to manage case sensitive and we can create enums for error codes

File: src/application/application-roles/application-roles.service.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          description,
+          name,
+          isDefault,
+          isSuperRole,
+          applicationsId,
+        },
+      });
+      this.logger.log('New role added!', newRole);
+      return {
+        message: 'successfully created a new role',
```
Comment (BackendBananeWala) [inline src/application/application-roles/application-roles.service.ts:38]: Please use the new response format provided

File: src/application/application-roles/application-roles.service.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    } catch (error) {
+      this.logger.log('Error creating a new Role', error);
+      return {
+        message: 'error creating the role',
+        role: data,
+      };
+    }
+  }
+
+  async getRole() {}
```
Comment (BackendBananeWala) [inline src/application/application-roles/application-roles.service.ts:51]: Throw not ye Implemented

File: src/application/application-scopes/application-scopes.service.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    } catch (error) {
+      this.logger.log('Error creating a new Scope', error);
+      return {
+        message: 'error creating the Scope',
+        scope: data,
+      };
+    }
+  }
+
+  async getScope() {}
```
Comment (BackendBananeWala) [inline src/application/application-scopes/application-scopes.service.ts:55]: Not yet implemented

File: src/application/application-scopes/application-scopes.service.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+          name,
+        },
+      });
+      this.logger.log('scope updated', scope);
+      return {
+        message: 'scope updated successfully',
+        scope,
+      };
+    } catch (error) {
+      console.log('Error occured while updating scope', error);
```
Comment (BackendBananeWala) [inline src/application/application-scopes/application-scopes.service.ts:101]: Console

File: src/main.ts [hunk 7]
```diff
@@ -20,16 +20,17 @@ async function bootstrap() {
...
     .setTitle('OIDC Protocol setup')
     .setDescription('This is the project to implement OIDC protocol in nestjs')
     .setVersion('1.0')
     .build();
   const document = SwaggerModule.createDocument(app, config);
   SwaggerModule.setup('api', app, document);
-  // saving the swagger file 
-  fs.writeFileSync('./swagger-spec.json', JSON.stringify(document, null, 2), { encoding: 'utf8' });
+  // saving the swagger file
+  fs.writeFileSync('./swagger-spec.json', JSON.stringify(document, null, 2), {
```
Comment (BackendBananeWala) [inline src/main.ts:31]: Use writeFile only if we are not awaiting for creation response. Also wrap in try catch to log error if occurred

**Pull request closed without merging**